Cython/Compiler/Options.py

```diff
@@ -176,6 +176,17 @@ def copy_inherited_directives(outer_directives, **new_directives):
     new_directives_out.update(new_directives)
     return new_directives_out
 
+def get_conversion_utility_code_directives(directives):
+    directives_out = dict(_directive_defaults)
+    allowed_directives = (
+        'binding', 'always_allow_keywords', 'allow_none_for_extension_args',
+        'auto_pickle', 'ccomplex',
+        'c_string_type', 'c_string_encoding')
+    for k,v in directives.items():
+        if k in allowed_directives or k.startswith('optimize.'):
+            directives_out[k] = v
+    return directives_out
```

Comment:
I would suggest
a) making this a global (list) constant, next to the defaults and scopes, since those three are what we need to adapt together and consistently as part of future changes in the directives.
b) make the list complete, including the spelled out `optimize` names, to make it more visible that they are part of what is being copied and make them show up in name searches.
c) just iterating over the name list and not the items, since the list to copy is much smaller than the directives dict.